gems/smithy/lib/smithy/templates/client/client.erb

```diff
@@ -52,6 +57,25 @@ module <%= module_name %>
       Smithy::Client::Input.new(handlers: handlers, context: context)
     end
 
+    def waiter(waiter_name, options = {})
+      waiter_class = waiters[waiter_name]
+      if waiter_class
+        waiter_class.new(options.merge(client: self))
+      else
+        raise Smithy::Client::Waiters::Errors::NoSuchWaiterError.new(waiter_name, waiters.keys)
+      end
+    end
+
+    def waiters
+<% if waiters? -%>
```

Comment:
Rather than check this, you should just have the view handle the empty case I think.